src/match-cache.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+const ROUTE_FALLBACK = '@ROUTE_FALLBACK';
```

Comment:
`forRoute` looks to be optional for `<RelativeFragment />`, why would you use `<RelativeFragment/>` without a `forRoute` prop? I wasn't sure why it was optional, so I just implemented a fallback identifier that would be matched when `forRoute` is falsy. Not sure if that's what we'd want here.